Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -76,12 +76,8 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
     /// To observe changes in the connection status, create an instance of `CurrentChatUserController`, and use it to receive
     /// callbacks when the connection status changes.
     ///
-    public var connectionStatus: ConnectionStatus {
-        ConnectionStatus(
-            webSocketConnectionState: webSocketClient?.connectionState ?? .disconnected(error: .ClientIsNotInActiveMode())
-        )
-    }
-    
+    public private(set) var connectionStatus: ConnectionStatus = .initialized
```

Comment:
It reflects the web-socket since we change the `connectionStatus` each time the `webSocketClient(_, didUpdateConectionState)` is invoked.

We also discussed that `connectionStatus` should be changed from `initialized` -> `disconnected` only when the current user session is restored but if we do:

```swift
public var connectionStatus: ConnectionStatus {
 guard let webSocket = webSocketClient else { .initilized }
 return ConnectionStatus(
  webSocketConnectionState: webSocketClient.connectionState
 )
}
```

for **active** client it becomes `disconnected` right after `init` is finished (instead of `initilized` which tells `client.currentUserId` and `currentUserController.currentUser?.id` can be different)